architecture/routing.adoc

```diff
@@ -128,77 +193,140 @@ Re-encryption termination is configured by setting `tls:termination` to `reencry
+To remove a passphrase from a keyfile, you can run:
 
-At the time of writing, a template router is the single type of router plugin available in OpenShift. 
+****
+`# openssl rsa -in _passwordProtectedKey.key_ -out _new.key_`
+****
 
-=== The HAProxy Template Router
+When creating a secure route, you must include your certificate files as a
+single line of text. Replace the existing line breaks with:
 
-The HAProxy template router implementation is the reference implementation for a template router plugin. This uses `openshift/origin-haproxy-router` to run an HAProxy instance alongside the template router plugin. To test routes, an install script is provided in `hack/install-router.sh`.
+****
+`\\n`
+****
 
-The route script requires two parameters, the router ID and the full URL to the master, and attempts to create the router based on the generated json file if it can find the `osc` executable on the path. If it cannot find the executable it creates the json file and notifies the user of the location. You can then manually run the create command.
+Note the double slash, which is required by the JSON spec.
 
+== Routers
+A template router provides certain infrastructure information to the underlying
+router implementation, such as:
+
+- A wrapper that watches endpoints and routes.
+- Endpoint and route data, which is saved into a consumable form.
+- Passing the internal state to a configurable template and executing the
+template.
+- Calling a reload script.
+
+Router plug-ins assume they can bind to host ports 80 and 443. This is to allow
+external traffic to route to the host and subsequently through the router.
+Routers also assume that networking is configured such that it can access all
+pods in the cluster.
+
+At the time of writing, a template router is the single type of router plug-in
+available in OpenShift.
+
+[[haproxy-template-router]]
+*HAProxy Template Router*
+
+The HAProxy template router implementation is the reference implementation for a
+template router plug-in. This uses the `openshift/origin-haproxy-router`
+repository to run an HAProxy instance alongside the template router plug-in. To
+test routes, an install script is provided in the *_hack/install-router.sh_*
+file.
+
+The router script requires two parameters: the router ID and the full URL to the
+master. It attempts to create the router based on the generated JSON file if it
+can find the `osc` executable on the path. If it cannot find the executable, it
+creates the JSON file and displays the location. You can then manually run the
+create command.
+
+.Running the Router Script
 ====
 
 ----
-[vagrant@openshiftdev origin]$ hack/install-router.sh router https://10.0.2.15:8443
+$ hack/install-router.sh router https://10.0.2.15:8443
 Creating router file and starting pod...
 router
 ----
 
 ====
 
-==== Data Flow
+The following diagram illustrates how data flows from the master through the
+plug-in and finally into a HAProxy configuration:
```

Comment:
Oops! Fixed.

Re: yellow line, you may have just clicked/highlighted the line at the time... ?